server/lib/realtime/subscription_manager.ex

```diff
@@ -0,0 +1,130 @@
+defmodule Realtime.SubscriptionManager do
+  use GenServer
+  require Logger
+
+  alias Realtime.RLS.Subscriptions
+  alias RealtimeWeb.Endpoint
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  def init(opts) do
+    case Keyword.fetch!(opts, :replication_mode) do
+      "RLS" = mode ->
+        sync_interval = Keyword.fetch!(opts, :subscription_sync_interval)
+
+        Endpoint.broadcast_from!(self(), "subscription_manager", "sync_subscription", nil)
+
+        ref =
+          Process.send_after(
+            self(),
+            :sync_subscription,
+            sync_interval
+          )
+
+        {:ok,
+         %{
+           replication_mode: mode,
+           sync_ref: ref,
+           sync_interval: sync_interval,
+           subscription_params: %{}
+         }}
+
+      "STREAM" = mode ->
```

Comment:
this makes it backwards compatible since both existing realtime and walrus/realtime shares the same `RealtimeChannel` and client now passing `user_token` when initiating channel topic